Automattic/wp-calypso
# Store: Remove remaining dependencies on old countries lib

Fixes #22062
See also #21574 

Design:
- Removes what remains of the old US and CA countries lib
- Switches us over to completely rely on the location data from the site itself instead
- Passes the countries as a prop into AddressView
- Make sure all users of AddressView isEditable handle country changes correctly

To Test:

- Delete any products on your test site (important - we still use this to decide whether or not to push defaults for currency, dimensions, etc)
- Optionally clear address and currency related options by installing https://github.com/allendav/woocommerce-reset-currency-and-country (to clear the options, simply visit the page that plugin adds to the Tools menu in wp-admin) 
- Use the WordPress.com console to reset `set_store_address_during_initial_setup` to 0 for your test site
- Navigate to http://calypso.localhost:3000/store/{DOMAIN}
- Set your store to a US address
- Go to /wp-admin/admin.php?page=wc-settings
- Make sure you see the US address, USD, currency pos: left, thousands: "," decimals: "." number of decimals: 2
- Go to /wp-admin/admin.php?page=wc-settings&tab=products
- Make sure you see ounces, inches
- Reset `set_store_address_during_initial_setup` to 0 again

- Set your store to a UK address (no state!)
- Go to /wp-admin/admin.php?page=wc-settings
- Make sure you see the UK address, Pounds Sterling, currency pos: left, thousands: "," decimals: "." number of decimals: 2
- Go to /wp-admin/admin.php?page=wc-settings&tab=products
- Make sure you see kilograms, centimeters
- Reset `set_store_address_during_initial_setup` to 0 again

- Go to Settings > Payments
- Make sure you can switch the store location back and forth between countries that have states and countries that do not
- Go to Settings > Shipping
- Make sure you can switch the address back and forth between countries that have states and countries that do not
- Go to Settings > Taxes
- Make sure that everything renders without error as you move the store location setting
- Go to Order > Edit Order > Customer Billing Address and Customer Shipping Address
- Make sure you can switch the store location back and forth between countries that have states and countries that do not
